sci-mathematics/mathematica/mathematica-13.1.0.ebuild

```diff
@@ -0,0 +1,182 @@
+# Copyright 1999-2022 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+CHECKREQS_DISK_BUILD=19G
+inherit check-reqs desktop unpacker xdg
+
+DESCRIPTION="Wolfram Mathematica"
+SRC_URI="
+	doc? ( WLDocs_${PV}_LINUX.sh )
+	Mathematica_${PV}_LINUX.sh
+"
+HOMEPAGE="https://www.wolfram.com/mathematica/"
+
+LICENSE="all-rights-reserved"
+KEYWORDS="-* ~amd64"
+SLOT="0"
+IUSE="cuda +doc ffmpeg R"
+
+RESTRICT="strip mirror bindist fetch"
+
+# Mathematica comes with a lot of bundled stuff. We should place here only what we
+# explicitly override with LD_PRELOAD.
+# RLink (libjri.so) requires dev-lang/R
+# FFmpegTools (FFmpegToolsSystem-5.0.so) requires media-video/ffmpeg
+RDEPEND="
+	cuda? ( dev-util/nvidia-cuda-toolkit )
+	media-libs/freetype
+	ffmpeg? ( media-video/ffmpeg )
+	R? ( dev-lang/R )
+	virtual/libcrypt
+"
+
+DEPEND="
+	${RDEPEND}
+"
+
+BDEPEND="
+	dev-util/patchelf
+"
+
+# we need this a few times
+MPN="Mathematica"
+MPV=$(ver_cut 1-2)
+M_BINARIES="MathKernel Mathematica WolframKernel wolframscript math mathematica mcc wolfram"
+M_TARGET="opt/Wolfram/${MPN}/${MPV}"
+
+# we might as well list all files in all QA variables...
+QA_PREBUILT="opt/*"
+
+S=${WORKDIR}
+
+src_unpack() {
+	/bin/sh "${DISTDIR}/Mathematica_${PV}_LINUX.sh" --nox11 --keep --target "${S}/unpack_app" -- "-help" || die
+	/bin/sh "${DISTDIR}/WLDocs_${PV}_LINUX.sh" --nox11 --keep --target "${S}/unpack_doc" -- "-help" || die
```

Comment:
This can be behind an `if use doc; then` I think. No need to unpack the documentation if we don't want it installed to the system. Besides, currently the ebuild will fail with `USE="-doc" because in the unpack phase we try to unpack a file that we haven't downloaded when `USE="-doc"`.

Reply:
Done